sbin/Makefile.am

```diff
@@ -55,10 +52,65 @@ if ENABLE_VNETBUILD
 scripts += vnetbuild
 endif
 
+
+if FILENAME_VERSIONS
+
+install-exec-hook:
+if ENABLE_FIREHOL
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f firehol firehol-@PACKAGE_VERSION@ && \
+		$(LN_S) firehol-@PACKAGE_VERSION@ firehol
+endif
+if ENABLE_FIREQOS
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f fireqos fireqos-@PACKAGE_VERSION@ && \
+		$(LN_S) fireqos-@PACKAGE_VERSION@ fireqos
+endif
+if ENABLE_LINK_BALANCER
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f link-balancer link-balancer-@PACKAGE_VERSION@ && \
+		$(LN_S) link-balancer-@PACKAGE_VERSION@ link-balancer
+endif
+if ENABLE_UPDATE_IPSETS
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f update-ipsets update-ipsets-@PACKAGE_VERSION@ && \
+		$(LN_S) update-ipsets-@PACKAGE_VERSION@ update-ipsets
+endif
+if ENABLE_VNETBUILD
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f vnetbuild vnetbuild-@PACKAGE_VERSION@ && \
+		$(LN_S) vnetbuild-@PACKAGE_VERSION@ vnetbuild
+endif
```

Comment:
Sorry that should have been: if the program is run using the link I think $0 will refer to the directory with the link, not the directory with the actual file.